src/main/kotlin/biokotlin/cli/MutateProteins.kt

```diff
@@ -0,0 +1,363 @@
+                }
+                start = range.end + 1
+            }
+        if (start < seqLength) {
+            result.add(BedfileRecord(contig, start, seqLength - 1))
+        }
+
+        return result
+
+    }
+
+    /**
+     * This deletes bases (number specified by parameter length) from each sequence.
+     * The bases are continuous. It makes the deletion in or out of
+     * the ranges defined by the bedfile depending on the setting of putMutationsInRanges.
+     */
+    private fun deletionMutation(
+        record: ProteinSeqRecord, ranges: Multimap<String, BedfileRecord>
+    ): String {
+
+        val origSeq = record.seq()
+        val seqLength = origSeq.length
+
+        val contig = record.id
+        val rangesForRecord = ranges.get(contig)
+
+        val mergedRanges = validateAndMergeRanges(seqLength, rangesForRecord)
+
+        val possibleMutationRanges =
+            if (putMutationsInRanges) mergedRanges else inverseRanges(contig, seqLength, mergedRanges)
+
+        // get possible ranges to mutate
+        // that are long enough to delete specified
+        // length. Then randomly select one of these ranges
+        val rangeToMutate = possibleMutationRanges
+            .filter { range -> range.length() >= length }
+            .randomOrNull()
+
+        val result = StringBuilder(origSeq)
+
+        if (rangeToMutate != null) {
+            // get random start index of place to delete
+            // within rangeToMutate
+            val start = Random(4321).nextInt(rangeToMutate.start, rangeToMutate.end - length + 2)
+
+            // delete length bases
+            result.delete(start, start + length)
+        }
+
+        return result.toString()
+
+    }
+
+    /**
+     * This inserts bases (number specified by parameter length) into each sequence.
+     * The bases are continuous. It makes the insertion in or out of
+     * the ranges defined by the bedfile depending on the setting of putMutationsInRanges.
+     */
+    private fun insertionMutation(
+        record: ProteinSeqRecord, ranges: Multimap<String, BedfileRecord>
+    ): String {
+
+        val origSeq = record.seq()
+        val seqLength = origSeq.length
+
+        val contig = record.id
+        val rangesForRecord = ranges.get(contig)
+
+        val mergedRanges = validateAndMergeRanges(seqLength, rangesForRecord)
+
+        val possibleMutationRanges =
+            if (putMutationsInRanges) mergedRanges else inverseRanges(contig, seqLength, mergedRanges)
+
+        // get possible ranges to mutate
+        // Then randomly select one of these ranges
+        val rangeToMutate = possibleMutationRanges
+            .randomOrNull()
+
+        val result = StringBuilder(origSeq)
+
+        if (rangeToMutate != null) {
+            // get random start index of place to insert
+            // within rangeToMutate
+            val start = Random(4321).nextInt(rangeToMutate.start, rangeToMutate.end + 1)
```

Comment:
It might be a better idea to make the seed an optional parameter the user can provide in case they want a different randomization.

Reply:
Agree with @zrm22 - this will also lead to better reproducibility.

Reply:
Does the same seed produce the exact same random numbers?

Reply:
Ok, I added that parameter

Reply:
It does when you do something like the following:
`val int1 = Random(4321).nextInt(100)`
`val int2 = Random(4321).nextInt(100)`

But you potentially have different start and end bounds so the following will be different:
`val int1 = Random(4321).nextInt(10,100)`
`val int2 = Random(4321).nextInt(5,95)`